src/solvers/lowering/byte_operators.cpp

```diff
@@ -17,20 +19,106 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/replace_symbol.h>
 #include <util/simplify_expr.h>
 
-#include "flatten_byte_operators.h"
+static exprt unpack_rec(
+  const exprt &src,
+  bool little_endian,
+  const exprt &offset,
+  const exprt &max_bytes,
+  const namespacet &ns,
+  bool unpack_byte_array = false);
+
+/// rewrite an array or vector into its individual bytes
+/// \param src: array/vector to unpack
+/// \param src_size: array/vector size
+/// \param element_width: bit width of array/vector elements
+/// \param little_endian: true, iff assumed endianness is little-endian
+/// \param offset: if not nil, use as lower bound for non-nil bytes to return
+/// \param max_bytes: if not nil, use as upper bound of the number of bytes to
+///                   unpack
+/// \param ns: namespace for type lookups
+/// \param [out] dest_array: target array_exprt to hold unpacked elements
+static void unpack_array_vector(
+  const exprt &src,
+  const exprt &src_size,
+  const mp_integer &element_width,
+  bool little_endian,
+  const exprt &offset,
+  const exprt &max_bytes,
+  const namespacet &ns,
+  array_exprt &dest_array)
+{
+  mp_integer num_elements;
+  if(to_integer(max_bytes, num_elements) && to_integer(src_size, num_elements))
+  {
+    throw "cannot unpack array/vector of non-const size:\n" +
+      src.type().pretty();
+  }
+
+  mp_integer offset_int = 0;
+
+  if(element_width > 0 && element_width % 8 == 0)
+  {
+    mp_integer el_bytes = element_width / 8;
+
+    if(num_elements % el_bytes != 0)
+      num_elements += el_bytes;
+
+    num_elements /= el_bytes;
```

Comment:
Now it's a number of elements. Please don't use something named `_elements` to store a byte count

Reply:
Fixed.